cfme/automate/dialogs/dialog_element.py

```diff
@@ -0,0 +1,238 @@
+        self.browser.drag_and_drop(self.element_loc(dragged_element),
+                                   self.element_loc(dropped_element))
+        view.save_button.click()
+        view = self.create_view(DetailsDialogView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+
+
+@attr.s
+class ElementCollection(BaseCollection):
+
+    ENTITY = Element
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def create(self, element_data):
+        for element in element_data:
+            view = navigate_to(self, "Add")
+            dragged_element = element.get('element_information').get('choose_type')
+            view.dd.drag_and_drop(dragged_element, self.parent.box_label)
+            view.fill(element.get('element_information'))
+            if element.get('options') is not None:
+                view.Options.fill(element.get('options'))
+            if element.get('advanced') is not None:
+                view.Advanced.fill(element.get('advanced'))
+            view.ele_save_button.click()
+            view.flash.assert_no_error()
+        view.save_button.click()
+        view = self.create_view(DialogsView)
+        assert view.is_displayed
+        view.flash.assert_message('{} was saved'.
+                                  format(self.parent.tab.dialog.label))
+        return self.instantiate(element_data=element_data)
+
+    def set_element_type(self, view, element):
+        """ Method to add element type.Depending on their type the subfields varies.
+
+        Args:
+            each_element: subfields depending on element type.
+        """
+        choose_type = element.get("choose_type")
+        dynamic_chkbox = element.get("dynamic_chkbox")
+        element_type = ['Dropdown', 'Radio Button']
+        if choose_type in element_type:
+            if not dynamic_chkbox:
+                row = view.entry_table.row(Value='<New Entry>')
+                row.click()
+                view.fill({'entry_value': "Yes",
+                           'entry_description': "entry_desc"})
+                view.add_entry_button.click()
+            else:
+                node1 = "InspectMe"
+                view.fill({'field_entry_point': 'b'})
+                view.bt_tree.click_path("Datastore", "new_domain", "System", "Request", node1)
+                view.apply_btn.click()
+                view.fill({'field_show_refresh_button': True})
+        if choose_type == "Text Area Box":
+            view.fill({'text_area': 'Default text'})
+
+
+@navigator.register(ElementCollection)
+class Add(CFMENavigateStep):
+    VIEW = AddElementView
+
+    prerequisite = NavigateToAttribute('parent.parent', 'Add')
+
+    def step(self):
+        self.prerequisite_view.dd.drag_and_drop("Text Box", self.obj.parent.box_label)
+
+
+@navigator.register(Element, 'Edit')
+class Edit(CFMENavigateStep):
+    VIEW = EditElementView
+
+    prerequisite = NavigateToAttribute('dialog', 'Edit')
```

Comment:
This navigation destination needs a step method defined, otherwise its unecessary as the prerequisite is the same page.

Reply:
Done